src/background/background.ts

```diff
@@ -15,33 +11,54 @@ chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
     if (message.command === "startRecordingBytesTransferred") {
         IBytesRepository.instance.clearBytesTransferred();
 
-        chrome.webRequest.onBeforeRequest.addListener(
-            catchPostRequestBodySize,
-            { urls: ["<all_urls>"], tabId },
-            ["requestBody"]
-        );
-        chrome.webRequest.onBeforeSendHeaders.addListener(
-            catchRequestHeaderSize,
-            { urls: ["<all_urls>"], tabId },
-            ["requestHeaders"]
-        );
-        chrome.webRequest.onCompleted.addListener(
-            catchResponseSize,
-            { urls: ["<all_urls>"], tabId },
-            ["responseHeaders"]
-        );
+        chrome.debugger.attach({ tabId: tabId }, "1.2", function () {
+            chrome.debugger.sendCommand(
+                { tabId: tabId },
+                "Network.enable",
+                {},
+                () => {
+                    if (chrome.runtime.lastError) {
+                        console.error(chrome.runtime.lastError);
+                    }
+                }
+            );
+        });
+
         sendResponse(true);
     }
 
     if (message.command === "stopRecordingBytesTransferred") {
-        chrome.webRequest.onCompleted.removeListener(catchResponseSize);
-        chrome.webRequest.onBeforeRequest.removeListener(
-            catchPostRequestBodySize
-        );
-        chrome.webRequest.onBeforeSendHeaders.removeListener(
-            catchRequestHeaderSize
-        );
+        try {
+            await chrome.debugger.detach({ tabId: tabId });
+        } catch (e: unknown) {
+            if (e instanceof Error) {
+                if (
+                    e.message ===
+                    `Debugger is not attached to the tab with id: ${tabId}.`
+                ) {
+                    console.warn(
+                        `Tried to detach debugger from tab (tabId: ${tabId}) when there was none attached. This is expected when a calculation starts, but should not be expected when a calculation is stopped.`
+                    );
+                }
+            }
+        }
         sendResponse(true);
     }
     return true;
 });
+
+chrome.debugger.onEvent.addListener(function (source, method, params) {
+    switch (method) {
+        case "Network.loadingFinished": {
+            const { encodedDataLength } = params as {
+                encodedDataLength?: number;
+            };
```

Comment:
Ew type casting - could you not just type the function at the top like so: `function (source: unknown, method: string, params: { encodedDataLength?: number}) { ... }`

Reply:
good catch!